### PR TITLE
Oleobj for office2007

### PR DESCRIPTION
**Pull request by christian-intra2net**
As hinted at earlier, an extension for oleobj to dump/extract embedded files from more types

This branch has seen a few rebases since initially I overlooked oleobj's existence and created my own oledump to perform the same task (extract embedded file from office files). I extended it to find embedded files in ppt, which required a new parser for ppt (the ppt_record_parser, which will hopefully soon replace the previous ppt_parser). Only after finishing this I realized there is already oleobj doing the same thing, so I merged by oledump into oleobj.

The new ppt parser is based on reading records, much like the xls parser, so I created record_base.py to contain the code common to both files.

Did not touch rtfobj which does the same thing on rtf.

### CODE REVIEW

**Comment by christian-intra2net** [comment]
Will add a few more test cases and self-made samples, but should not change the code outside tests dir

File: oletools/msodde.py [hunk 1]
```diff
@@ -67,10 +67,14 @@
...
+try:
+    from oletools.thirdparty import olefile
+except ImportError:
+    PARENT_DIR = dirname(dirname(abspath(__file__)))
+    if PARENT_DIR not in sys.path:
+        sys.path.insert(0, PARENT_DIR)
+    del PARENT_DIR
+    from oletools.thirdparty import olefile
 
 from oletools.thirdparty import olefile
```
Comment (decalage2) [inline oletools/msodde.py]: this one is redundant, can be removed.

Comment (christian-intra2net) [reply oletools/msodde.py]: I guess you mean the one on line 75/79? I'll send you another pull request for this

**Comment by christian-intra2net** [comment]
Will add a few more test cases and self-made samples, but should not change the code outside tests dir

**Comment by christian-intra2net** [comment]
That was fast! Thank you :-)

**Pull request merged**